zerver/lib/markdown/__init__.py

```diff
@@ -1829,13 +1829,19 @@ def handleMatch(self, m: Match[str], data: str) -> Tuple[Optional[Element], Any,
         return None, None, None
 
 
-class UserGroupMentionPattern(markdown.inlinepatterns.Pattern):
-    def handleMatch(self, m: Match[str]) -> Optional[Element]:
-        match = m.group(2)
+class UserGroupMentionPattern(markdown.inlinepatterns.InlineProcessor):
+    def handleMatch(self, m: Match[str], data: str) -> Tuple[Optional[Element], Any, Any]:
+        match = m.group(0)
 
         db_data = self.md.zulip_db_data
         if self.md.zulip_message and db_data is not None:
-            name = extract_user_group(match)
+            # As InlineProcessor API sends all the matching part
+            # coming after (?<![^\s\'\"\(,:<]), so @ is also
+            # included in match so it needs to be strip.
+            if match.startswith("@*") and match.endswith("*"):
+                name = match[2:-1]
```

Comment:
I think it'd be better to write this with variables, with `start_syntax = "@*"`, and then using that variable (E.g. len(start_syntax):-len(end_syntax)`) here -- we try to avoid numbers like this.

Reply:
Made these changes!